src/git.js

```diff
@@ -159,6 +159,8 @@ class Git {
 		// split diff into separate entries for separate files. \ndiff --git should be a reliable way to detect the separation, as content of files is always indented
 		return `\n${ string }`.split('\ndiff --git').slice(1).reduce((resultDict, fileDiff) => {
 			const lines = fileDiff.split('\n')
+			if (lines.length === 0) return resultDict; // ignore binary files
```

Comment:
I'm not sure if it's the best way to fix it in the long term. But I hope this will at least prevent from crashing halfway and not syncing all files. Please let me know what you think